src/index.coffee

```diff
@@ -6,45 +6,39 @@ BrowserTransport = require './transports/browser-transport'
 CliTransport = require './transports/cli-transport'
 Logger = require './logger'
 
-
-getGlobalScope = ->
-  if typeof global != 'undefined'
-    return global
-
-  return window
-
-
-isBrowser = (globalScope) ->
-  return (
-    globalScope.window?.document?.createElement and
-    globalScope.window?.name != 'nodejs'
-  )
+{ getGlobalScope, isBrowser } = require './utils'
+{ setupTransport } = require './setup-transport'
 
 
 module.exports =
   LogLevels: LogLevels
   
   Logger: Logger
 
+  BrowserTransport: BrowserTransport
+  CliTransport: CliTransport
+
   create: (options = {}) ->
     globalScope = getGlobalScope()
     browser = isBrowser(globalScope)
 
     if browser
-      transport = new BrowserTransport(globalScope.console)
-      transport.setLogLevelPrefixes(prefixes.browser)
-      transport.setLogLevelStyles(colors.browser)
+      output = globalScope.console
+      transportSource = BrowserTransport
+
     else
-      transport = new CliTransport(globalScope.process.stderr)
-      transport.setLogLevelPrefixes(prefixes.cli)
-      transport.setLogLevelColors(colors.cli)
+      output = globalScope.process.stderr
+      transportSource = CliTransport 
 
-    logger = new Logger(transport, options)
+    transport = if options.transport then new options.transport(output) else new transportSource(output)
```

Comment:
I would definitely prefer a transport instance being passed in rather than the constructor.

```
transport = options.transport or new DefaultTransport(output)
```

Reply:
I see. This was originally to work with our software. Since I'll be exporting the transports now it can be the instance. Will fix this.

Reply:
The only disadvantage is that you have to be passing the instance that used its own "output" in constructor. But I guess this is allright since the concerns are separated here.